Kuroba/app/src/main/java/com/github/adamantcheese/chan/features/embedding/embedders/RjcodeEmbedder.java

```diff
@@ -0,0 +1,108 @@
+package com.github.adamantcheese.chan.features.embedding.embedders;
+
+import android.graphics.Bitmap;
+import android.util.JsonReader;
+
+import com.github.adamantcheese.chan.BuildConfig;
+import com.github.adamantcheese.chan.core.model.PostImage;
+import com.github.adamantcheese.chan.core.net.NetUtilsClasses;
+import com.github.adamantcheese.chan.core.repository.BitmapRepository;
+import com.github.adamantcheese.chan.features.embedding.EmbedResult;
+import com.github.adamantcheese.chan.utils.StringUtils;
+import com.google.common.io.Files;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import okhttp3.HttpUrl;
+
+public class RjcodeEmbedder
+        extends JsonEmbedder {
+
+    // Matches strings of the form RJ123456. Does not match if the string is
+    // preceded by a "/", since that is handled by DlsiteEmbedder instead.
+    private final Pattern RJCODE_PATTERN =
+            Pattern.compile("\\w*(?<!\\/)[rR][jJ](\\d{6})(?![^ .,?!:;)\\]'\"\\-=+_/\\n])");
```

Comment:
Is there a reason for the long "not followed by" regex? Does `(?<!\\/)[rR][jJ](\\d{6})` not work by itself?

Reply:
I included the "not followed by" regex because the rjcode regex casts a pretty wide net as opposed to the other embedders looking for `http://...`, so I didn't want to accidentally capture something that isn't an rjcode. I was thinking that it's possible for some randomized link or encoded message to include `...rj123456...` as part of a larger string, where it'd be wrong to attempt to replace it. So I searched through a bunch of threads and added in any characters after rjcodes I saw and any other ones I could think of. If you think it's unnecessary though, it can definitely be removed.